frontend/src/Components/PieChart/getPieChartVoteData.ts

```diff
@@ -0,0 +1,57 @@
+import { ChartDataset } from 'chart.js';
+import { CardValue, VOTE_COFFEE, VOTE_NOTE_VOTED, VOTE_OBSERVER } from '../../../../shared/cards';
+import { Votes } from '../../../../shared/serverMessages';
+import { compareCardValues } from '../ResultsPage/compareVotes';
+
+export const getPieChartVoteData = (
+  votes: Votes
+): { labels: CardValue[]; datasets: ChartDataset<'pie', number[]>[] } => {
+  const accumulatedVotes = Object.entries(
+    Object.values(votes).reduce(
+      (
+        votesByValue: Partial<Record<CardValue, number>>,
+        value: CardValue
+      ): Partial<Record<CardValue, number>> => {
+        if ([VOTE_OBSERVER, VOTE_NOTE_VOTED, VOTE_COFFEE].includes(value)) {
+          return votesByValue;
+        }
+
+        return {
+          ...votesByValue,
+          [value]: (votesByValue[value] ?? 0) + 1,
+        };
+      },
+      {}
+    )
+  ) as [CardValue, number][];
+
+  accumulatedVotes.sort(([value1], [value2]) => compareCardValues(value1, value2));
```

Comment:
Not a big fan of reduce as it tends to be slightly hard to read (and inefficient, though this does not matter here). Why not just a simple `for..of` loop to the same effect, then you can also avoid the spread operator?

```suggestion
  const accumulatedVotesByValue: { [vote in CardValue]?: number } = {};
  for (const vote of Object.values(votes)) {
    if (![VOTE_OBSERVER, VOTE_NOTE_VOTED, VOTE_COFFEE].includes(vote)) {
      accumulatedVotesByValue[vote] = (accumulatedVotesByValue[vote] ?? 0) + 1;
    }
  }
  const accumulatedVotes = (Object.entries(accumulatedVotesByValue) as [CardValue, number][]).sort(
    ([value1], [value2]) => compareCardValues(value1, value2)
  );
```

Reply:
Done in 3e33abf